src/Command/TemplateCommand.php

```diff
@@ -270,13 +270,18 @@ protected function _loadController(): array
 
         $namespace = Configure::read('App.namespace');
 
-        $primaryKey = (array)$modelObject->getPrimaryKey();
-        $displayField = $modelObject->getDisplayField();
-        $singularVar = $this->_singularName($this->controllerName);
-        $singularHumanName = $this->_singularHumanName($this->controllerName);
-        $schema = $modelObject->getSchema();
-        $fields = $schema->columns();
-        $modelClass = $this->modelName;
+        try {
+            $primaryKey = (array)$modelObject->getPrimaryKey();
+            $displayField = $modelObject->getDisplayField();
+            $singularVar = $this->_singularName($this->controllerName);
+            $singularHumanName = $this->_singularHumanName($this->controllerName);
+            $schema = $modelObject->getSchema();
+            $fields = $schema->columns();
+            $modelClass = $this->modelName;
+        } catch (\Exception $exception) {
+            echo $exception->getMessage() . "\n";
+            exit;
```

Comment:
exit makes the whole thing not testable. also no error code but success code 0 here.
so this is currently worse than before.

Reply:
Please can you provide constructive feedback. You've provided no requested changes or route to move forward.

Reply:
It was - you just have to either output a 1 exit code (see status code constants in Command class) - or throw some "AbortException", I suspect the code also offers some abort() convenience wrapper for this.

Reply:
Great, thanks. That's much better than 'this is currently worse than before'

Reply:
You can use `$this->abort()` here.